### PR TITLE
[DOCS] Adds HuggingFace example to inference API tutorial

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds an example of using the inference API with the Hugging Face service to the semantic search tutorial.

### Preview

[Semantic search inference](https://elasticsearch_bk_107298.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/semantic-search-inference.html)

### CODE REVIEW

File: docs/reference/search/search-your-data/semantic-search-inference.asciidoc [hunk 1]
```diff
@@ -6,9 +6,11 @@
...
 The instructions in this tutorial shows you how to use the {infer} API with
 various services to perform semantic search on your data. The following examples 
-use Cohere's `embed-english-v3.0` model and OpenAI's `text-embedding-ada-002`
-second generation embedding model. You can use any Cohere and OpenAI models,
-they are all supported by the {infer} API.
+use Cohere's `embed-english-v3.0` model, the `all-mpnet-base-v2` model from
+HuggingFace, and OpenAI's `text-embedding-ada-002` second generation embedding
+model. You can use any Cohere and OpenAI models, they are all supported by the
+{infer} API. For a list of supported models available on HuggingFace, refer to
+{ml-docs}/ml-nlp-model-ref.html[the compatible third party NLP models] page.
```
Comment (jonathan-buttner) [inline docs/reference/search/search-your-data/semantic-search-inference.asciidoc:13]: Hmm, I think the models that we support for hugging face via the inference api is different than what we have listed in that `ml-nlp-model-ref` page.

The ones I've tested are these:

https://huggingface.co/sentence-transformers/all-MiniLM-L12-v2
https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2
https://huggingface.co/intfloat/multilingual-e5-base
https://huggingface.co/intfloat/e5-base-v2
https://huggingface.co/sentence-transformers/all-mpnet-base-v2
https://huggingface.co/intfloat/e5-small-v2
https://huggingface.co/intfloat/multilingual-e5-small

Others _may_ work, but we haven't tested them. And they'd need to adhere to the response format that we're expecting, so we should probably list these specific models somewhere as the ones we support right now.

File: docs/reference/search/search-your-data/semantic-search-inference.asciidoc [hunk 2]
```diff
@@ -91,7 +93,8 @@ GET _tasks/<task_id>
 // TEST[skip:TBD]
 
 You can also cancel the reindexing process if you don't want to wait until the
-reindexing process is fully complete which might take hours:
+reindexing process is fully complete which might take hours for data sets this
```
Comment (jonathan-buttner) [inline docs/reference/search/search-your-data/semantic-search-inference.asciidoc:96]: nit: maybe `which might take hours for large data sets:`

**Review by jonathan-buttner — COMMENTED** [review]
Looks good, I left a few suggestions

File: docs/reference/search/search-your-data/semantic-search-inference.asciidoc (hunk 1 above)
Comment (szabosteve) [reply docs/reference/search/search-your-data/semantic-search-inference.asciidoc:13]: Precious info, thank you! In an upcoming commit, I'll add the list to the PUT inference API page as part of this PR.

File: docs/reference/tab-widgets/inference-api/infer-api-search.asciidoc [hunk 3]
```diff
@@ -72,6 +73,79 @@ query from the `cohere-embeddings` index sorted by their proximity to the query:
...
+  "_source": [
+    "id",
+    "content"
+  ]
+}
+--------------------------------------------------
+// TEST[skip:TBD]
+
+As a result, you receive the top 10 documents that are closest in meaning to the
+query from the `cohere-embeddings` index sorted by their proximity to the query:
```
Comment (szabosteve) [inline docs/reference/tab-widgets/inference-api/infer-api-search.asciidoc:102]:
```suggestion
query from the `hugging-face-embeddings` index sorted by their proximity to the query:
```

File: docs/reference/tab-widgets/inference-api/infer-api-search.asciidoc [hunk 4]
```diff
@@ -24,7 +24,8 @@ GET cohere-embeddings/_search
 // TEST[skip:TBD]
 
 As a result, you receive the top 10 documents that are closest in meaning to the
-query from the `cohere-embeddings` index sorted by their proximity to the query:
+query from the `cohere-embeddings` index sorted by their proximity to the
+query:
```
Comment (szabosteve) [inline docs/reference/tab-widgets/inference-api/infer-api-search.asciidoc:28]:
```suggestion
query from the `cohere-embeddings` index sorted by their proximity to the query:
```

**Review by jonathan-buttner — APPROVED** [review]
Thanks for the changes!

**Pull request merged**